cmd/hypper/upgrade.go

```diff
@@ -114,6 +122,12 @@ func newUpgradeCmd(cfg *action.Configuration, logger log.Logger) *cobra.Command
 						logger.Info(eyecandy.ESPrintf(settings.NoEmojis, "Release %q does not exist. Installing it now. :toolbox:\n", client.ReleaseName))
 					}
 					instClient := action.NewInstall(cfg)
+					// Set namespace for the install client
```

Comment:
we have almost identical code in three places now (twice here line 125-130, 168-174 and in install.go) can we move that to a static helper function?

Reply:
I agree. I would schedule that inside https://github.com/rancher-sandbox/hypper/issues/70 (and rename the card to "fix name and namespace setting deduplication" or so).

Reply:
I would prefer to see all duplications everywhere, to see the pattern, and then fix them in a way that is good for all callers, hence why I want to postpone that.

Reply:
Agreed, card updated!

Reply:
> I would prefer to see all duplications everywhere, to see the pattern, and then fix them in a way that is good for all callers, hence why I want to postpone that.

Well, why wait? I suggest using a helper package with static functions that can be invoked wherever needed... That can't hurt. If we see a pattern we can still move that function later. But we have it in place now.

Reply:
I think this is out of scope for this, as this deals with a bug which needs to be fixed, not an improvement.

Reply:
I'm personally happy with the solution for now, we have the issue open to not forget about it.

Reply:
Ok, then we should as this as new issue.

Reply:
got a PR already for this :D